openassessment/xblock/static/js/src/lms/oa_response.js

```diff
@@ -307,6 +311,32 @@ export class ResponseView {
       return this.baseView.buttonEnabled('.step--response__submit', enabled);
     }
 
+    /**
+     * Disable the fields to avoid any changes while an action is processing.
+     * For example, use this when beginning a file upload, or submitting the response.
+     */
+    disableFields() {
+      this.submitEnabled(false);
+
+      const responseEl = $('.step--response', this.element);
+      responseEl.find('.file__upload').prop('disabled', true);
```

Comment:
@samuelallan72 

I am wondering if using a jQuery class selector to select the button class directly would be better here than selecting a parent class and then traversing to find the button. Not sure if there is performance trade-off in either way.

Also, if go the selector route, we already have a handy helper method [this.baseView.buttonEnabled](https://github.com/openedx/edx-ora2/blob/master/openassessment/xblock/static/js/src/lms/oa_base.js#L512) which we can use.

Reply:
Ah nice, thanks for the suggestions. Both of these sound like improvements. I was simply following the existing patterns in the file.

Reply:
Updated, thanks :)